### PR TITLE
Login request not working

### PR DESCRIPTION
**Pull request by ko-yaroslav**

### CODE REVIEW

File: server/utils/createError.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const createError = ({ message, transKey, path, type}) => {
+    return {
+        message,
+        transKey,
+        details: [
+            {
+                path,
+                type,
+                message
```
Comment (yaroslavfedyshyn) [inline server/utils/createError.js]: @ko-yaroslav it is not correct. There shouldn't be general error message in each detail. I think we don't need message here at all as we don't use it anywhere.

File: server/routes/auth/resetPassword/_post.js [hunk 2]
```diff
@@ -1,37 +1,36 @@
...
 const User = require('../../../models/user');
 const UserActions = require('../../../models/userActions');
+const createError = require('../../../utils/createError');
 
 module.exports = async (req, res, next) => {
-  try {
-    const userActions = await UserActions.findOne({ token: req.body.token });
+    const userActions = await UserActions.findOne({token: req.body.token});
+
+    const errData = {
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/resetPassword/_post.js]: @ko-yaroslav only message and transKey should be here. No need for some details

File: server/utils/createError.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const createError = ({ message, transKey, path, type}) => {
+    return {
+        message,
+        transKey,
+        details: [
```
Comment (yaroslavfedyshyn) [inline server/utils/createError.js]: @ko-yaroslav what if I want to add more then 1 detail? How should I do it?

File: server/routes/auth/login/_post.js [hunk 4]
```diff
@@ -2,15 +2,28 @@ const Session = require('../../../models/session');
...
     email,
     password,
   } = req.body;
 
-  try {
-    const user = User.findOne({ email }, async (err, user) => {
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      const errData = {
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/login/_post.js]: @ko-yaroslav only message and transKey should be present here

File: server/routes/auth/login/_post.js [hunk 5]
```diff
@@ -26,7 +26,19 @@ module.exports = async (req, res, next) => {
 
       user.comparePassword(password, async (err, isMatch) => {
         if (err) throw err;
-        if (isMatch) {
+        if(!isMatch) {
+          const errData = {
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/login/_post.js]: @ko-yaroslav only message and transKey should be present. no need for details

File: server/utils/createError.js (hunk 1 above)
Comment (ko-yaroslav) [reply server/utils/createError.js]: Done.

File: server/utils/createError.js (hunk 3 above)
Comment (ko-yaroslav) [reply server/utils/createError.js]: Done.

File: server/routes/auth/resetPassword/_post.js (hunk 2 above)
Comment (ko-yaroslav) [reply server/routes/auth/resetPassword/_post.js]: Done.

File: server/utils/createError.js [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
-                message
-            }
-        ]
+const createError = ({generalMessage, transKey, ...options}) => {
+
+    if (Object.keys(options).length) {
+        return {
+            message: generalMessage,
+            transKey,
+            details: [options]
```
Comment (yaroslavfedyshyn) [inline server/utils/createError.js]: @ko-yaroslav make possible to pass more than 1 detail to details object

**Comment by yaroslavfedyshyn** [comment]
@ko-yaroslav go through all of my comments

File: server/routes/auth/login/_post.js (hunk 4 above)
Comment (ko-yaroslav) [reply server/routes/auth/login/_post.js]: Done.

File: server/routes/auth/login/_post.js (hunk 5 above)
Comment (ko-yaroslav) [reply server/routes/auth/login/_post.js]: Done.

File: server/utils/createError.js (hunk 6 above)
Comment (ko-yaroslav) [reply server/utils/createError.js]: Done.

File: server/routes/auth/resetPassword/_post.js [hunk 7]
```diff
@@ -9,7 +9,7 @@ module.exports = async (req, res, next) => {
 
     const errData = {
         generalMessage: 'not found',
-        transKey: 'server.error.something.wrong'
+        transKey: 'Something went wrong'
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/resetPassword/_post.js]: @ko-yaroslav trans key should not be just a text. It should be key that matches frontend key.

File: server/routes/auth/login/_post.js [hunk 8]
```diff
@@ -17,7 +17,7 @@ module.exports = async (req, res, next) => {
         path:   ['email'],
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/login/_post.js]: @ko-yaroslav remove path and type out of here

File: server/routes/auth/login/_post.js [hunk 9]
```diff
@@ -31,7 +31,7 @@ module.exports = async (req, res, next) => {
             path:   ['email'],
```
Comment (yaroslavfedyshyn) [inline server/routes/auth/login/_post.js]: @ko-yaroslav remove path and type out of here

File: server/routes/auth/resetPassword/_post.js (hunk 7 above)
Comment (ko-yaroslav) [reply server/routes/auth/resetPassword/_post.js]: Done.

File: server/routes/auth/login/_post.js (hunk 8 above)
Comment (ko-yaroslav) [reply server/routes/auth/login/_post.js]: Done.

File: server/routes/auth/login/_post.js (hunk 9 above)
Comment (ko-yaroslav) [reply server/routes/auth/login/_post.js]: Done.

**Comment by yaroslavfedyshyn** [comment]
@ko-yaroslav fix conflicts

**Comment by ko-yaroslav** [comment]
> @ko-yaroslav fix conflicts

Done.

**Pull request merged**